internal/config/certs.go

```diff
@@ -85,6 +85,9 @@ func LoadX509KeyPair(certFile, keyFile string) (tls.Certificate, error) {
 	if len(rest) > 0 {
 		return tls.Certificate{}, ErrSSLUnexpectedData(nil).Msg("The private key contains additional data")
 	}
+	if key == nil || key.Headers == nil {
```

Comment:
```
// A Block represents a PEM encoded structure.
//
// The encoded form is:
//    -----BEGIN Type-----
//    Headers
//    base64-encoded Bytes
//    -----END Type-----
// where Headers is a possibly empty sequence of Key: Value lines.

type Block struct {
        Type    string            // The type, taken from the preamble (i.e. "RSA PRIVATE KEY").
        Headers map[string]string // Optional headers.
        Bytes   []byte            // The decoded bytes of the contents. Typically a DER encoded ASN.1 structure.
}
```

it looks like key.Headers is optional, let's not error out if it is nil

Reply:
This is correct. PEM headers are optional.

Reply:
> ```
> // A Block represents a PEM encoded structure.
> //
> // The encoded form is:
> //    -----BEGIN Type-----
> //    Headers
> //    base64-encoded Bytes
> //    -----END Type-----
> // where Headers is a possibly empty sequence of Key: Value lines.
> 
> type Block struct {
>         Type    string            // The type, taken from the preamble (i.e. "RSA PRIVATE KEY").
>         Headers map[string]string // Optional headers.
>         Bytes   []byte            // The decoded bytes of the contents. Typically a DER encoded ASN.1 structure.
> }
> ```
> 
> it looks like key.Headers is optional, let's not error out if it is nil

There is a reason why we error out, observe the code.

The subsequent code requires that Headers to be non nil. It is not like this PR checks if key.Headers is empty it checks if it's uninitialized.